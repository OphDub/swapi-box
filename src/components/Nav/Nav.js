import React from 'react';
import './Nav.css';
import { NavLink } from 'react-router-dom';
import { func } from 'prop-types';

const Nav = ({ getData }) => {
  return (
    <section className="nav-section">
      <label htmlFor="reveal-email" className="btn"> Menu </label>
      <input type="checkbox" id="reveal-email" role="button"></input>
      <nav id="nav-category" className="categories">
        <NavLink to='/people' className="nav-category-links">
          <button onClick={() => getData('people')} className="nav-category-btns">
            People
          </button>
        </NavLink>
        <NavLink to='/planets' className="nav-category-links">
          <button onClick={() => getData('planets')} className="nav-category-btns">
            Planets
          </button>
        </NavLink>
        <NavLink to='/vehicles' className="nav-category-links">
          <button onClick={() => getData('vehicles')} className="nav-category-btns">
            Vehicles
          </button>
        </NavLink>
      </nav>
    </section>
  );
};

Nav.propTypes = {
  getData: func.isRequired
};

export default Nav;