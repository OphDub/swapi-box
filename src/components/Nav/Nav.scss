.nav-section {
  display: flex;
  flex-direction: column;
}

.nav-category {
  display: flex;
  flex-direction: column;
  min-width: 100%;
}

button {
  border: #FFFFFF 2px solid;
  color: #FFFFFF;
  font-size: 1em;
  background-color: #000000;
  padding: 0.75em 2em;
}

button:hover {
  background: #41b9e1;
}

input[type=checkbox] {
  display: none;
}

input[type=checkbox]:checked~.remove-check {
  display: none;
}

.btn {
  background: none;
  border: 2px solid #FFFFFF;
  cursor: pointer;
  display: none;
  text-align: center;
  margin: 0;
  padding: 10px 20px 11px;
  text-decoration: none;
}

.btn:hover {
  background: #41b9e1;
}

@media (max-width: 400px) {
  .nav-section {
    width: 160px;
  }

  #nav-category {
    display: none;
  }

  .nav-category-links {
    display: block;
  }

  .nav-category-btns {
    width: 100%;
  }

  .btn {
    display: inline-block;
  }

  input[type=checkbox]:checked~#nav-category {
    display: block;
  }
}