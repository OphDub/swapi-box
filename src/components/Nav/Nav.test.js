/* eslint-disable */
import React from 'react';
import Nav from './Nav';

describe('NAV', () => {
  it('should match the snapshot', () => {

  });

  it('has buttons which should call getData on click', () => {

  });
});