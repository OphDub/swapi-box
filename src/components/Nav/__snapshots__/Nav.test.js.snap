// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NAV should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/people"
>
          <button
                    onClick={[Function]}
          >
                    People
          </button>
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/planets"
>
          <button
                    onClick={[Function]}
          >
                    Planets
          </button>
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/vehicles"
>
          <button
                    onClick={[Function]}
          >
                    Vehicles
          </button>
</NavLink>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <button
            onClick={[Function]}
>
            People
</button>,
          "to": "/people",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "People",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <button
            onClick={[Function]}
>
            Planets
</button>,
          "to": "/planets",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Planets",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <button
            onClick={[Function]}
>
            Vehicles
</button>,
          "to": "/vehicles",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/people"
>
            <button
                        onClick={[Function]}
            >
                        People
            </button>
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/planets"
>
            <button
                        onClick={[Function]}
            >
                        Planets
            </button>
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/vehicles"
>
            <button
                        onClick={[Function]}
            >
                        Vehicles
            </button>
</NavLink>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <button
              onClick={[Function]}
>
              People
</button>,
            "to": "/people",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "People",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "People",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <button
              onClick={[Function]}
>
              Planets
</button>,
            "to": "/planets",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Planets",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Planets",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <button
              onClick={[Function]}
>
              Vehicles
</button>,
            "to": "/vehicles",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Vehicles",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Vehicles",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
